### ISSUE TITLE
Can't install Symfony 4

### ISSUE THREAD

**Issue by plbertret** [flo-sch/stripe-bundle#7]
Hi !
(and thank you for this bundle)

Seems that the bundle is not yet compatible with Symfony 4

flosch/stripe-bundle 0.1.9 requires symfony/symfony ~3.0 -> satisfiable by symfony/symfony[3.0.x-dev, 3.1.x-dev, 3.2.x-dev, 3.3.x-dev, 3.4.x-dev, v3.0.0, v3.0.0-BETA1, v3.0.1, v3.0.2, v3.0.3, v3.0.4, v3.0.5, v3.0.6, v3.0.7, v3.0.8, v3.0.9, v3.1.0, v3.1.0-BETA1, v3.1.0-RC1, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.0-BETA1, v3.2.0-RC1, v3.2.0-RC2, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13, v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.0-BETA1, v3.3.0-RC1, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.14, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.0-BETA1, v3.4.0-BETA2, v3.4.0-BETA3, v3.4.0-BETA4, v3.4.0-RC1, v3.4.0-RC2, v3.4.1] but these conflict with your requirements or minimum-stability.

**Comment by flo-sch** [comment]
Hey @plbertret, yeah, most likely, I was waiting for the release, didn't had time to dig into it last month :)

Unfortunately, I'm affraid I won't be able to check SF4 before next week... Unless you have the time to make a Pull-Request which I would be glad to merge.

I'll let you know when I can release a SF4-ready version!

**Comment by plbertret** [comment]
I think I still don't have the skills to take this responsibility but take your time ;)

**Comment by flo-sch** [comment]
Okay thanks for reminding btw, I'll try to do it tonight or tomorrow... Hope there's no actual incompatibility and just editing composer.json file will be enough!

**Comment by ekeyte** [comment]
+1 Definitely would love this change!

**Comment by flo-sch** [comment]
Can any of you test it with the brand new 2.0.0 release, thanks to @plbertret for the changes?
`composer require flosch/stripe-bundle 2.0.0`

**Comment by ekeyte** [comment]
Thanks! I will this evening.

**Comment by iCodr8** [comment]
It's not working. The problem is that you define a specific version of `stripe-php` in the composer file. This should be done by the project, where this bundle is used.

So change it to:

```
"stripe/stripe-php": "*"
```

**Comment by ekeyte** [comment]
Would changing the stripe-php dependency to be any version potentially break the configuration files offered by this library?

**Comment by flo-sch** [comment]
@iCodr8: to answer you the reason behind this:

> The problem is that you define a specific version of stripe-php in the composer file.

the problem of allowing any version of such an external dependency is that people would then assume that it would work for `any` version of `stripe/stripe-php`, now, and in the future...

As I do not contribute to `stripe/stripe-php` myself, I cannot make such an assumption on an external dependency.
It might work for now for all the versions `>= 3.0`, but I haven't tested ;
And I cannot assume that any new major version of it would not require any adjustment to this bundle.

Which is the reason why I allowed any `3.XXX` version at the time of writing, this was by design.

> This should be done by the project, where this bundle is used.

Why so? Even though it is possible of course.
The purpose of this bundle is to make the `stripe/stripe-php` integration easier.
Requiring the library itself from the project seems anti-pattern to me (but I might be wrong on that point).

However, what is possible to do is to allow more versions of `stripe/stripe-php` -- as soon as such versions can be tested`stripe/stripe-php` can be tested.
I did not had time to do it myself so far though, but if you can test it and make a PR, I would gladly merge that either into the v1 or v2 version of this bundle :)

**Comment by flo-sch** [comment]
@ekeyte I do not think it would break the symfony configuration files themselves, but the risk is that it could break the integration if the API offered from `stripe/stripe-php` has changed...

Which is why you want to use semantic versionning in the first place.
